File: Sample/2.0/Models/Sample/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyApp.Models.Sample
{
  [Table("Orders", Schema = "dbo")]
  public class Order
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id
    {
      get;
      set;
    }


    [InverseProperty("Order")]
    public ICollection<OrderDetail> OrderDetails { get; set; }
    public DateTime OrderDate
    {
      get;
      set;
    }
    public string UserName
    {
      get;
      set;
    }
  }
}

File: Sample/2.0_VB/Models/Order.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace MyApp.Models.Sample
    <Table("Orders", Schema:="dbo")>
    Public Class Order
        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Id() As Integer

        <InverseProperty("Order")>
        Public Property OrderDetails() As ICollection(Of OrderDetail)

        Public Property OrderDate() As DateTime

        Public Property UserName() As String

    End Class
End Namespace
